valhalla/jawn/src/lib/shared/request/request.ts

```diff
@@ -200,26 +185,38 @@ export async function getRequestsCached(
 
   const requests = await dbExecute<HeliconeRequest>(query, builtFilter.argsAcc);
 
-  if (!supabaseServer) {
-    return resultMap(requests, (data) => {
-      return truncLargeData(data, MAX_TOTAL_BODY_SIZE);
-    });
-  }
-
-  const rosettaWrapper = new RosettaWrapper(supabaseServer);
-  const results = await mapLLMCalls(requests.data, rosettaWrapper);
+  const results = await mapLLMCalls(requests.data, s3Client, orgId);
 
   return resultMap(results, (data) => {
-    return truncLargeData(data, MAX_TOTAL_BODY_SIZE);
+    return data;
   });
 }
 
 async function mapLLMCalls(
   heliconeRequests: HeliconeRequest[] | null,
-  rosettaWrapper: RosettaWrapper
+  s3Client: S3Client,
+  orgId: string
 ): Promise<Result<HeliconeRequest[], string>> {
   const promises =
     heliconeRequests?.map(async (heliconeRequest) => {
+      // First retrieve s3 signed urls if past the implementation date
+      const s3ImplementationDate = new Date("2024-03-21");
```

Comment:
wait why is this hard coded? lol

Reply:
We want to go to S3 if it is stored in S3; otherwise, we won't get the signed URL. Prevents unnecessary calls.

There are other ways to do this, but this is pretty easy.

Reply:
yes, added timestamp, thanks for reminder